### DISCUSSION TITLE
How to handle external and internal IP addresses

### DISCUSSION THREAD

**Discussion by Cursky** [waylaidwanderer/PandoraAI#132] [Q&A]
May I ask you, after running my program, two links appear: localhost:3001 and http://xx.xx.xx.xxx:3001/. The xx represents my internal network IP, and my cloud server can only be accessed through an external IP address. How can I change xx.xx.xx.xxx to my external IP address?

**Comment by lover-shadow** [comment]
“I can access xxx.xx.xx:3001 on the internal network from the outside using NAT, but I can only open the page and not respond. Everything is normal on the internal network. What could be the problem?

**Comment by timefox** [comment]
It's not a good idea to expose the original 'node-chatgpt-api' server on internet. Because it lacks a verification mechanism, this means that anyone can access it without restrictions.
But if you really want to do this, you can:
1.find out ${config.apiBaseUrl} in 'components/Chat.vue', replace it with your 'external IP address'. you need to modify the setting of 'node-chatgpt-api' server also, to make it run at the same 'external IP address'.
or
2. find out ${config.apiBaseUrl} in 'components/Chat.vue', replace it with a redirect path like '/chatapi/'. and then configure the reverse proxy on the server to access it.

**Comment by waylaidwanderer** [comment]
> It's not a good idea to expose the original 'node-chatgpt-api' server on internet. Because it lacks a verification mechanism, this means that anyone can access it without restrictions.

It's fine if you don't set an API key and have the user do it themselves via PandoraAI instead, though :)
